modules/programs/getmail-accounts.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, ... }:
+
+with lib;
+
+{
+  options.getmail = {
+    enable = mkEnableOption "the getmail mail retriever for this account";
+
+    destinationCommand = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "\${pkgs.maildrop}/bin/maildrop";
+      description = ''
+        Specify a command delivering the incoming mail to your maildir.
+      '';
+    };
+
+    mailboxes = mkOption {
+      type = types.listOf types.str;
+      default = [];
```

Comment:
If the user must supply at least one value I would suggest to leave this empty and set the type to `types.nonEmptyListOf types.str`. Then the assertion in `programs/getmail.nix` should be unnecessary.

Reply:
Aaah nice! I was wondering about such a type. Taking that, I now found the `lib/types.nix` module.

Reply:
Removed the assertion. The only way I could make it fail now is by not specifying an `imap` submodule and/or some of it's attributes. Is there anything to guard against without expensive `assertions`? Or ignore that for now?

Reply:
The mbsync module contains
https://github.com/rycee/home-manager/blob/8991fe2e90a9b35a98a8b98f02c9de5425b11785/modules/programs/mbsync.nix#L147-L163
which you can copy if you want but I think you can skip it for now. I'll try to put together a more general solution to make these assertions as soon as possible.

Reply:
Cheers @rycee. Should have looked into the module before commenting. I'll be happy to skip it for now and might add it with another patch for the `IDLE` support. I suppose there is no 100% prevention to stuff up the `imap` configuration to end up in an error.